napari_console/qt_console.py

```diff
@@ -181,24 +179,33 @@ def _capture(self):
             if "NAPARI_EMBED" not in c.frame.f_globals:
                 self.push(dict(c.namespace))
 
-    def _update_theme(self, event=None):
+    def _update_theme(self, event=None, style_sheet=''):
         """Update the napari GUI theme."""
-        from napari.utils.theme import get_theme, template
         from napari.qt import get_stylesheet
+        from napari.utils.theme import get_theme, template
 
-        # qtconsole unfortunately won't inherit the parent stylesheet
-        # so it needs to be directly set
-        raw_stylesheet = get_stylesheet()
-        # template and apply the primary stylesheet
-        # (should probably be done by napari)
-        # After napari 0.4.11, themes are evented models rather than
-        # dicts.
+        # After napari 0.5.0 the `as_dict` kwarg has been deprecated
+        # and will be removed in a future version.
         theme = get_theme(self.viewer.theme, as_dict=True)
-        self.style_sheet = template(raw_stylesheet, **theme)
-
-        # After napari 0.4.6 the following syntax will be allowed
-        # self.style_sheet = get_stylesheet(self.viewer.theme)
+        # In the future, with napari 0.5.0+, the following syntax will be allowed
+        # theme = get_theme(self.viewer.theme).to_rgb_dict()
 
+        # qtconsole unfortunately won't inherit the parent stylesheet
+        # so it needs to be directly set when required.
+        if style_sheet:
+            # napari 0.5.x uses the `style_sheet` kwarg
+            self.style_sheet = style_sheet
+        else:
+            # napari 0.4.x doesn't use the `style_sheet` kwarg
+            raw_stylesheet = get_stylesheet()
+            # template and apply the primary stylesheet
+            # (should probably be done by napari)
+            # After napari 0.4.11, themes are evented models rather than
+            # dicts.
+            self.style_sheet = template(raw_stylesheet, **theme)
+
+            # After napari 0.4.6 the following syntax will be allowed
+            # self.style_sheet = get_stylesheet(self.viewer.theme)
```

Comment:
The imports and `get_theme` call should move int else clause.

Reply:
Thanks for rechecking this! Latests commits should change the imports placement. Also noticed that still there was a need to connect to the viewer theme events when no style sheet is being passed (napari <0.5.5/without the changes done over napari/napari#6753). Re-added that event connection logic in case no style sheet is detected over the constructor.

Let me know if there is something else that needs to be done here!